be/src/exec/spill/spiller.h

```diff
@@ -127,39 +128,39 @@ class Spiller : public std::enable_shared_from_this<Spiller> {
     // no thread-safe
     // TaskExecutor: Executor for runing io tasks
     // MemGuard: interface for record/update memory usage in io tasks
-    template <class TaskExecutor, class MemGuard>
-    Status spill(RuntimeState* state, const ChunkPtr& chunk, TaskExecutor&& executor, MemGuard&& guard);
+    template <class MemGuard>
+    Status spill(RuntimeState* state, const ChunkPtr& chunk, MemGuard&& guard);
```

Comment:
We should not couple spiller with executor. The specific scheduling rules should be controlled by the executor. If we want to control the task behavior we can consider providing a new implementation of the executor

Reply:
If you want to simplify the input parameters, you can implement a caller similar to the one in UT to wrap the spiller.